### DISCUSSION TITLE
Recreating the Phanta DB and adding custom sequences

### DISCUSSION THREAD

**Discussion by nickp60** [bhattlab/phanta#16] [Phanta DBs] [answered]
Hi Phanta team!  We need to add some sequences to the phanta db, namely the exact genomes from a mock community quality control sample that we routinely run, and some spike we use.  Do you have a script for database creation that you could share?  Based on the methods in the paper, it looks like the bacterial/archaeal sequences are from HumGut?

In addition to the script for building the kraken db, could you share how `host_prediction_to_genus.tsv` and `species_name_to_vir_score.txt` were created? 

Thanks in advance!

Nick

**Comment by yipinto** [comment]
Hi Nick,
Thanks for reaching out. We would love to help.
Yes, bacterial and archaeal sequences are from HumGut.
host_prediction_to_genus.tsv was calculated using [iphop](https://bitbucket.org/srouxjgi/iphop/src/main/), but you can use any host prediction tool as long as you keep the format.
species_name_to_vir_score.txt was calculated based on BACPHLIP. 
For more information on both see the "Provided postprocessing scripts" section in Phanta's manuscript. 

As for scripts to add sequences to the database, @meenachakra will be able to help with the details here, as she helped with that to another Phanta user recently, but we would love to share any script we used to build the database and provide any help and guidance. It also depends on whether you need to add sequences to the viral(MGV+refseq), prokaryotic or other lineage and whether the genomes' species already exist in the db and if you want to dereplicate. Feel free to email us and we can discuss the details there.

**Reply by meenachakra** [reply]
Thanks Yishay!
Hi Nick, thanks for the question! I will respond in more detail within the week.

**Reply by nickp60** [reply]
Thanks so much!  If its helpful [here](https://github.com/vdblab/resources/blob/b0ff9a1f0b46ae70458ef120360ee379bacc6d4d/workflow/rules/krakenbracken.smk#L60) is the code I used to make a kraken/bracken db for the mock only.  I'll get the bugs processed with iphop and bacphlip.

**Reply by meenachakra** [reply]
Hi Nick! Thanks for sharing the code. 

Just to clarify, you'll need to add the genomes listed in the below (plus the spike)? What is the source of the genomes - RefSeq?

We already have genomes for a lot of these species in the Phanta DB but perhaps you can add the new genomes as strains (which we can help with). Thanks!

cat D6331.refseq/genomes/Akkermansia_muciniphila.fasta | sed "s,>,>kraken:taxid|239935,g" > D6331.refseq/tmp/Akkermansia_muciniphila.fasta
    cat D6331.refseq/genomes/Bacteroides_fragilis.fasta | sed "s,>,>kraken:taxid|817,g" > D6331.refseq/tmp/Bacteroides_fragilis.fasta
    cat D6331.refseq/genomes/Bifidobacterium_adolescentis.fasta | sed "s,>,>kraken:taxid|1680,g" > D6331.refseq/tmp/Bifidobacterium_adolescentis.fasta
    cat D6331.refseq/genomes/Candida_albican.fasta | sed "s,>,>kraken:taxid|5476y,g" > D6331.refseq/tmp/Candida_albican.fasta
    cat D6331.refseq/genomes/Clostridioides_difficile.fasta | sed "s,>,>kraken:taxid|1496,g" > D6331.refseq/tmp/Clostridioides_difficile.fasta
    cat D6331.refseq/genomes/Clostridium_perfringens.fasta | sed "s,>,>kraken:taxid|1502,g" > D6331.refseq/tmp/Clostridium_perfringens.fasta
    cat D6331.refseq/genomes/Enterococcus_faecalis.fasta | sed "s,>,>kraken:taxid|1351,g" > D6331.refseq/tmp/Enterococcus_faecalis.fasta
    cat D6331.refseq/genomes/Escherichia_coli_B1109.fasta | sed "s,>,>kraken:taxid|562,g" > D6331.refseq/tmp/Escherichia_coli_B1109.fasta
    cat D6331.refseq/genomes/Escherichia_coli_b2207.fasta | sed "s,>,>kraken:taxid|562,g" > D6331.refseq/tmp/Escherichia_coli_b2207.fasta
    cat D6331.refseq/genomes/Escherichia_coli_B3008.fasta | sed "s,>,>kraken:taxid|562,g" > D6331.refseq/tmp/Escherichia_coli_B3008.fasta
    cat D6331.refseq/genomes/Escherichia_coli_B766.fasta | sed "s,>,>kraken:taxid|562,g" > D6331.refseq/tmp/Escherichia_coli_B766.fasta
    cat D6331.refseq/genomes/Escherichia_coli_JM109.fasta | sed "s,>,>kraken:taxid|562,g" > D6331.refseq/tmp/Escherichia_coli_JM109.fasta
    cat D6331.refseq/genomes/Faecalibacterium_prausnitzii.fasta | sed "s,>,>kraken:taxid|853,g" > D6331.refseq/tmp/Faecalibacterium_prausnitzii.fasta
    cat D6331.refseq/genomes/Fusobacterium_nucleatum.fasta | sed "s,>,>kraken:taxid|851,g" > D6331.refseq/tmp/Fusobacterium_nucleatum.fasta
    cat D6331.refseq/genomes/Lactobacillus_fermentum.fasta | sed "s,>,>kraken:taxid|1613,g" > D6331.refseq/tmp/Lactobacillus_fermentum.fasta
    cat D6331.refseq/genomes/Methanobrevibacter_smithii.fasta | sed "s,>,>kraken:taxid|2173,g" > D6331.refseq/tmp/Methanobrevibacter_smithii.fasta
    cat D6331.refseq/genomes/Prevotella_corporis.fasta | sed "s,>,>kraken:taxid|28128,g" > D6331.refseq/tmp/Prevotella_corporis.fasta
    cat D6331.refseq/genomes/Roseburia_hominis.fasta | sed "s,>,>kraken:taxid|301301,g" > D6331.refseq/tmp/Roseburia_hominis.fasta
    cat D6331.refseq/genomes/Saccharomyces_cerevisiae.fasta | sed "s,>,>kraken:taxid|4932,g" > D6331.refseq/tmp/Saccharomyces_cerevisiae.fasta
    cat D6331.refseq/genomes/Salmonella_enterica.fasta | sed "s,>,>kraken:taxid|28901,g" > D6331.refseq/tmp/Salmonella_enterica.fasta
    cat D6331.refseq/genomes/Veillonella_rogosae.fasta | sed "s,>,>kraken:taxid|423477,g" > D6331.refseq/tmp/Veillonella_rogosae.fasta
    echo "Building DB"

**Reply by meenachakra** [reply]
Also, with regards to adding to host_prediction_to_genus.tsv and species_name_to_vir_score.txt, you will only need to do this if you are adding viruses to the DB - which you are not, correct? thank you :)

**Reply by nickp60** [reply]
Hi @meenachakra, yes, I need the specific strains to be represented in the database.  The genomes Zymo provides don't list the actual accesession, but as an example the PHANTA A. municiphila representative is Akkermansia muciniphila ATCC BAA-835.  Neither of the two genome assemblies ([ASM1750414v1](https://www.ncbi.nlm.nih.gov/assembly/GCF_017504145.1) or [ASM2022v1](https://www.ncbi.nlm.nih.gov/assembly/GCF_000020225.1)) match the A municiphila they provide ([link](https://s3.amazonaws.com/zymo-files/BioPool/D6331.refseq.zip))

Same with the spike in organisms -- the species are likely in the database, but I need the exact strains  to be there.

None are viruses, so sounds like we can skip adding things to host_prediction_to_genus or species_name_to_vir_score.txt, yay!

Thanks!

**Comment by meenachakra** [comment, marked as answer]
Great, thanks! You'll need the modify the original library/taxonomy files that we used to create the database and then rebuild the database. This should be fairly straightforward and we can do it together in a Zoom call, if that works. Please feel free to email us (https://www.bhattlab.com/people) so we can schedule.

**Reply by haolilan** [reply]
I am also puzzled by the construct of the library/species_genome_size.txt. Is there a script for this, or whom can I ask for help? Thank you very much.

**Reply by meenachakra** [reply]
Yes, please rename this from .txt to .py and use this Python script with the path to the database folder as the single argument. Let us know if it doesn't work for you or you have questions. This is the version that we use in-house and it works; however, we will likely polish it and add more comments before we post the DB-making tutorial. 
[calc_genome_size.txt](https://github.com/bhattlab/phanta/files/15221912/calc_genome_size.txt)

**Reply by haolilan** [reply]
Thank you very much. I'll give it a try and give a feedback. :)
